pkg/operator/cluster/ceph/osd/osd.go

```diff
@@ -197,13 +229,48 @@ func (c *Cluster) Start() error {
 				}
 			}
 		} else {
-			logger.Infof("osd replica set started for node %s", n.Name)
+			logger.Infof("osd prepare job started for node %s", n.Name)
 		}
 
 		// wait for the current node's orchestration to be completed
-		if err := c.waitForCompletion(n.Name); err != nil {
-			errorMessages = append(errorMessages, err.Error())
+		if status, err := c.waitForCompletion(n.Name); err != nil {
+			logger.Warningf("failed to prepare node %s: %v", n.Name, err)
+			err = discover.FreeDevices(c.context, n.Name, clusterName)
+			if err != nil {
+				logger.Warningf("failed to free devices: %s", err)
+				errorMessages = append(errorMessages, err.Error())
+			}
 			continue
+		} else {
+			// start osds
+			osds := status.OSDs
+			logger.Debugf("osds prepared on node %s: %+v", n.Name, osds)
+			for _, osd := range osds {
+				logger.Debugf("start osd %v", osd)
+				dp := c.makeOSDDeployment(n.Name, devicesToUse, n.Selection, resources, osd)
```

Comment:
should `makeOSDDeployment()` return an error? there might be multiple reasons for the deployment to not be created instead of assuming it is because of empty volumes?

Reply:
this just returns the api object. the error will be detected once the api object is to be created.